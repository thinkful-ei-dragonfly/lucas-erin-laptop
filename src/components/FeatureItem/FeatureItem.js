import React from 'react';

function Features(props){
  const featureItems = props.features.map(item => {
    return (
      <div key={item.key} className={item.props.className}>
        <div className='feature__name'>{item.key}</div>
        <ul className='feature__list'>
          {item.props.children[1].props.children}
        </ul>
      </div>
    )
  })

  return (
    <div>
    {featureItems}
    </div>)
}

export default Features;
