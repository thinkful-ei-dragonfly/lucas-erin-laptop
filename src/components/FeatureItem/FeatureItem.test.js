import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import FeatureItem from './FeatureItem';
import STORE from '../../STORE.js'
import App from '../../App'
configure({ adapter: new Adapter() });

describe('FeatureItem', ()=> {
  const features = Object.keys(STORE)
        .map(key => {
          const options = STORE[key].map((item, index) => {
            const selectedClass = item.name === STORE[key].name ? 'feature__selected' : '';
            const featureClass = 'feature__option ' + selectedClass;
            return <li key={index} className="feature__item">
              <div className={featureClass}

                onClick={e => App.updateFeature(key, item)}>
                  { item.name }
                  ({ new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD'})
                    .format(item.cost) })
              </div>
            </li>
          });

          return <div className="feature" key={key}>
            <div className="feature__name">{key}</div>
            <ul className="feature__list">
              { options }
            </ul>
          </div>
        });
  it('renders with enzyme', () => {
    const wrapper = shallow(<FeatureItem features={features}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it('simulates clicking 2nd li', () => {
    const wrapper = shallow(<FeatureItem features={features} />)
    wrapper.find('li').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('simulates multiple clicks', () => {
    const wrapper = shallow(<FeatureItem features={features} />)
    wrapper.find('li').at(1).simulate('click')
    wrapper.find('li').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
