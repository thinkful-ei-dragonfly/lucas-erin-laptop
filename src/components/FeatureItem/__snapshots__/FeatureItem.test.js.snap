// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureItem renders with enzyme 1`] = `
<div>
  <div
    className="feature"
    key="Processor"
  >
    <div
      className="feature__name"
    >
      Processor
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          17th Generation Intel Core HB (7 Core with donut spare)
          (
          $700.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Professor X AMD Fire Breather with sidewinder technology
          (
          $1,200.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Operating System"
  >
    <div
      className="feature__name"
    >
      Operating System
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Ubuntu Linux 16.04
          (
          $200.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Bodhi Linux
          (
          $300.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Video Card"
  >
    <div
      className="feature__name"
    >
      Video Card
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Toyota Corolla 1.5v
          (
          $1,150.98
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Mind mild breeze 2000
          (
          $1,345.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Display"
  >
    <div
      className="feature__name"
    >
      Display
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
          (
          $1,500.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.3" HGTV (3840 x 2160) Home makeover edition
          (
          $1,400.00
          )
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FeatureItem simulates clicking 2nd li 1`] = `
<div>
  <div
    className="feature"
    key="Processor"
  >
    <div
      className="feature__name"
    >
      Processor
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          17th Generation Intel Core HB (7 Core with donut spare)
          (
          $700.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Professor X AMD Fire Breather with sidewinder technology
          (
          $1,200.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Operating System"
  >
    <div
      className="feature__name"
    >
      Operating System
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Ubuntu Linux 16.04
          (
          $200.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Bodhi Linux
          (
          $300.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Video Card"
  >
    <div
      className="feature__name"
    >
      Video Card
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Toyota Corolla 1.5v
          (
          $1,150.98
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Mind mild breeze 2000
          (
          $1,345.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Display"
  >
    <div
      className="feature__name"
    >
      Display
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
          (
          $1,500.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.3" HGTV (3840 x 2160) Home makeover edition
          (
          $1,400.00
          )
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FeatureItem simulates multiple clicks 1`] = `
<div>
  <div
    className="feature"
    key="Processor"
  >
    <div
      className="feature__name"
    >
      Processor
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          17th Generation Intel Core HB (7 Core with donut spare)
          (
          $700.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Professor X AMD Fire Breather with sidewinder technology
          (
          $1,200.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Operating System"
  >
    <div
      className="feature__name"
    >
      Operating System
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Ubuntu Linux 16.04
          (
          $200.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Bodhi Linux
          (
          $300.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Video Card"
  >
    <div
      className="feature__name"
    >
      Video Card
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Toyota Corolla 1.5v
          (
          $1,150.98
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          Mind mild breeze 2000
          (
          $1,345.00
          )
        </div>
      </li>
    </ul>
  </div>
  <div
    className="feature"
    key="Display"
  >
    <div
      className="feature__name"
    >
      Display
    </div>
    <ul
      className="feature__list"
    >
      <li
        className="feature__item"
        key="0"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
          (
          $1,500.00
          )
        </div>
      </li>
      <li
        className="feature__item"
        key="1"
      >
        <div
          className="feature__option "
          onClick={[Function]}
        >
          15.3" HGTV (3840 x 2160) Home makeover edition
          (
          $1,400.00
          )
        </div>
      </li>
    </ul>
  </div>
</div>
`;
